File: my_jacket_api/src/main/java/com/example/service/CartService.java
package com.example.service;

import com.example.dto.ICartDetailDto;
import com.example.dto.ICartDto;
import com.example.repository.ICartDetailRepository;
import com.example.repository.ICartRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CartService implements ICartService {
    @Autowired
    private ICartRepository iCartRepository;
    @Autowired
    private ICartDetailRepository iCartDetailRepository;

    @Override
    public void createCart(Integer userId) {
        if (userId != null) {
            iCartRepository.createCart(userId);
        }else {
            iCartRepository.createCart(null);
        }
    }

    @Override
    public void createCartDetail(Integer quantity, Integer cartId, Integer productDetailId) {
        iCartDetailRepository.createCartDetail(quantity, cartId, productDetailId);
    }

    @Override
    public ICartDto getCartByUserId(Integer id) {
        return iCartRepository.getCartByUserId(id);
    }

    @Override
    public void insertOrUpdateCartDetail(Integer cartId, Integer productDetailId, Integer quantity, Integer userId) {
        iCartDetailRepository.insertOrUpdateCartDetail(cartId, productDetailId, quantity, userId);
    }

    @Override
    public void updateAmountCartDetail(Integer cartId, Integer productDetailId, Integer quantity, Integer userId) {
        iCartDetailRepository.updateAmountCartDetail(cartId, productDetailId, quantity, userId);
    }

    @Override
    public void delete(Integer userId, Integer cartId, Integer productId, Integer productDetailId) {
        iCartDetailRepository.delete(userId,cartId,productId,productDetailId);
    }

    @Override
    public void deleteCartDetailFlowInvoice(Integer userId, Integer cartId) {
        iCartDetailRepository.deleteCartDetailFlowInvoice(userId, cartId);
    }
}

File: my_jacket_react_app/my_jacket/src/service/CartService.js
import axios from "axios";

export const createCart = async (value) => {
  try {
    const respone = await axios.post(
      "http://localhost:8080/api/cart/new",
      value
    );
    return respone.status;
  } catch (error) {
    console.log(error);
  }
};

export const createCartDetail = async (value) => {
  try {
    const respone = await axios.post(
      "http://localhost:8080/api/cart/detail/new",
      value
    );
    return respone.status;
  } catch (error) {
    console.log(error);
  }
};
export const getCartByUserId = async (id) => {
  try {
    const respone = await axios.get("http://localhost:8080/api/cart/" + id);
    return respone.data.cart_id;
  } catch (error) {
    console.log(error);
  }
};

export const getCartDetail = async (accountId, cartId) => {
  try {
    const respone = await axios.get(
      `http://localhost:8080/api/cart/cart-detail/${accountId}/${cartId}`
    );
    return respone.data;
  } catch (error) {
    console.log(error);
  }
};

export const insertOrUpdateCartDetail = async (value) => {
  try {
    const respone = await axios.post(
      "http://localhost:8080/api/cart/cart-detail/new",
      value
    );
    return respone.status;
  } catch (error) {
    console.log(error);
  }
};

export const updateAmountCartDetail = async (value) => {
    try {
      const respone = await axios.post(
        "http://localhost:8080/api/cart/cart-detail/update-amount",
        value
      );
      return respone.status;
    } catch (error) {
      console.log(error);
    }
  };

  export const deleteCartDetail = async (userId,cartId,productId,productDetailId) => {
    try {
      const respone = await axios.delete(
        `http://localhost:8080/api/cart/cart-detail/${userId}/${cartId}/${productId}/${productDetailId}`);
      
      return respone.status;
    } catch (error) {
      console.log(error);
    }
  };

  export const deleteCartDetailFlowInvoice = async (userId,cartId) => {
    try {
      const respone = await axios.delete(
        `http://localhost:8080/api/cart/cart-detail/${userId}/${cartId}`);
      
      return respone.status;
    } catch (error) {
      console.log(error);
    }
  };


